iset.mm

```diff
@@ -64682,6 +64682,113 @@ would need to be strengthened (for example, to imply included middle) to
+        wral ad2antrl simpll simprr simprl sssucid syl5ss simplr fidcenumlemrks
+        mtbir wfo mp2and exp31 finds sylc ) AFUCOAFGPZQZHEFRZOZWBSZUDZLAVSMUEAU
+        AUFZGPZQZHEWERZOZWISZUDZUKATGPZQZHETRZOZWOSZUDZUKAUBUFZGPZQZHEWRRZOZXBS
+        ZUDZUKZAWRUGZGPZQZHEXFRZOZXJSZUDZUKVTWDUKUAUBFWETULZWGWMWKWQXMWFWLAWETG
+        UHUIXMWIWOWJWPXMWHWNHWETEUJUMZXMWIWOXNUNUOUPUAUBUQZWGWTWKXDXOWFWSAWEWRG
+        UHUIXOWIXBWJXCXOWHXAHWEWREUJUMZXOWIXBXPUNUOUPWEXFULZWGXHWKXLXQWFXGAWEXF
+        GUHUIXQWIXJWJXKXQWHXIHWEXFEUJUMZXQWIXJXRUNUOUPWEFULZWGVTWKWDXSWFVSAWEFG
+        UHUIXSWIWBWJWCXSWHWAHWEFEUJUMZXSWIWBXTUNUOUPWMWPWOWPWMWOHTOHURWNTHEUSUT
+        VMVAVBWRUCOZXEXHXLYAXEQZXHQZBCDEWRGHABCUQVCCDVDBDVDYBXGIVEAGUCOYBXGJVEA
+        GDEVNYBXGKVEYAXEXHVFYBAXGVGZYCAWSXDYBAXGVHYCWRXFGWRVIYDVJYAXEXHVKVOAHDO
+        YBXGNVEVLVPVQVR $.
+    $}
+
+    ${
+      $d A x y $.  $d F k w $.  $d F k x y $.  $d N k w $.  $d k ph w $.
+
+      $( Lemma for ~ fidcenum .  Reverse direction (put into deduction form).
+         (Contributed by Jim Kingdon, 19-Oct-2022.) $)
+      fidcenumlemr $p |- ( ph -> A e. Fin ) $=
+        ( cima cfn wceq syl wss wcel wa adantr cv wi c0 vw vk wfo ssid com csuc
+        foima anbi2d imaeq2 eleq1d imbi12d ima0 0fin eqeltri a1i cfv wn csn cun
+        sseq1 wfn simprl fofn simprr sucid sseldd fnsnfv syl2anc uneq2d imaeq2i
+        vex df-suc imaundi eqtri syl6eqr simpr snssd ssequn2 sylib sssucid sstr
+        eqtr3d mpan ad2antll simplr mp2and eqeltrd fof ffvelrnd unsnfi eqeltrrd
+        syl3anc wdc wral simpll fidcenumlemrk mpjaodan exp31 finds mpcom mpan2
+        wf ) AEFJZDKAFDEUCZXCDLIFDEUGMAFFNZXCKOZFUDFUEOZAXEPZXFAXGXEHQAUARZFNZP
+        ZEXIJZKOZSATFNZPZETJZKOZSAUBRZFNZPZEXRJZKOZSZAXRUFZFNZPZEYDJZKOZSXHXFSU
+        AUBFXITLZXKXOXMXQYIXJXNAXITFUTUHYIXLXPKXITEUIUJUKXIXRLZXKXTXMYBYJXJXSAX
+        IXRFUTUHYJXLYAKXIXREUIUJUKXIYDLZXKYFXMYHYKXJYEAXIYDFUTUHYKXLYGKXIYDEUIU
+        JUKXIFLZXKXHXMXFYLXJXEAXIFFUTUHYLXLXCKXIFEUIUJUKXQXOXPTKEULUMUNUOXRUEOZ
+        YCYFYHYMYCPZYFPZXREUPZYAOZYHYQUQZYOYQPZYGYAKYSYAYPURZUSZYGYAYOUUAYGLZYQ
+        YOUUAYAEXRURZJZUSZYGYOYTUUDYAYOEFVAZXRFOYTUUDLYOAUUFYNAYEVBZAXDUUFIFDEV
+        CMMYOYDFXRYNAYEVDXRYDOYOXRUBVKVEUOVFZFXREVGVHVIYGEXRUUCUSZJUUEYDUUIEXRV
+        LVJEXRUUCVMVNVOZQYSYTYANUUAYALYSYPYAYOYQVPVQYTYAVRVSWBYOYBYQYOAXSYBUUGY
+        EXSYNAXRYDNYEXSXRVTXRYDFWAWCWDZYMYCYFWEWFZQWGYOYRPZUUAYGKYOUUBYRUUJQUUM
+        YBYPDOZYRUUAKOYOYBYRUULQYOUUNYRYOFDXREYOXDFDEXBYOAXDUUGIMZFDEWHMUUHWIZQ
+        YOYRVPYAYPDWJWLWKYOBCDEXRFYPYOABRCRLWMCDWNBDWNUUGGMYOAXGUUGHMUUOYMYCYFW
+        OUUKUUPWPWQWRWSWTXAWK $.
+    $}
+  $}
+
+  ${
+    $d A f n x y $.
+    $( A set is finite if and only if it has decidable equality and is finitely
+       enumerable.  Proposition 8.1.11 of [AczelRathjen], p. 72.  (Contributed
+       by Jim Kingdon, 19-Oct-2022.) $)
+    fidcenum $p |- ( A e. Fin <-> ( A. x e. A A. y e. A DECID x = y
+        /\ E. n e. _om E. f f : n -onto-> A ) ) $=
```

Comment:
So I conclude that in iset.mm conventions, an enumeration allows repetitions ?  Is this standard ?
I haven't checked, but maybe it's safer to mention it at the first occurrence of "enumeration" or "enumerable" in iset.mm
Disregard if it's already the case.

Reply:
> So I conclude that in iset.mm conventions, an enumeration allows repetitions ?  Is this standard ?
> I haven't checked, but maybe it's safer to mention it at the first occurrence of "enumeration" or "enumerable" in iset.mm
> Disregard if it's already the case.

I don't think we have enough theorems of this sort to have a convention really. "Finitely enumerable" is as defined in AczelRathjen, maybe I should cite their definition explicitly because we have been bitten a bit about not noticing two (inequivalent, I think) definitions for "countable" as mentioned at https://github.com/metamath/set.mm/issues/2595#issuecomment-1287083072

If by "enumeration without repetitions" you mean a bijection, the word for having one from a natural number is "finite" (df-fin and df-en which I think matches other sources).

Reply:
So your reasoning is that if one reads "finitely enumerable", it has to mean something else than "finite", else we would use that latter expression. Therefore, it can only mean that repetitions are allowed (i.e., surjection from a finite ordinal, instead of the stronger requirement of a bijection). I agree, but this may not be totally obvious on first reading, so if this can be said at the point where it's first introduced in iset.mm, it would be nice.

Reply:
> maybe I should cite their definition explicitly

I have updated this pull request with that change.